### ISSUE TITLE
nested Instance exits proof mode

### ISSUE THREAD

**Issue by coqbot** [coq/coq#4638]
Note: the issue was created automatically with bugzilla2github tool

Original bug ID: BZ#4638
From: @siegebell
Reported version: 8.5
CC:   @gares, @JasonGross

See also: [BZ#3890](https://github.com/coq/coq/issues?q=is%3Aissue%20%22Original%20bug%20ID%3A%20BZ%233890%22)

**Comment by ejgallego** [comment]
@maximedenes can you see if this solved, and if not what remains to be done?

**Comment by gares** [comment]
The bug is in the code of VtUnknown, when we will remove it (8.10 or 8.11) this bug will go.
One can avoid the bug by not turning on the Refine instance Mode option.
But the code is till there if you look for it.

**Comment by ejgallego** [comment]
Still a problem cc : @maximedenes , the parsing state is not there:
```
egallego@lasserre:~/external/coq-master$ dune exec -- coqc k.v -debug
Error: Anomaly "Uncaught exception Option.IsNone."
Please report at http://coq.inria.fr/bugs/.
frame @ file "toplevel/coqc.ml", line 58, characters 4-16
frame @ file "toplevel/coqc.ml", line 37, characters 2-81
frame @ file "list.ml", line 106, characters 12-15
frame @ file "toplevel/ccompile.ml", line 188, characters 2-39
frame @ file "toplevel/ccompile.ml", line 127, characters 18-88
frame @ file "toplevel/vernac.ml", line 177, characters 30-88
raise @ file "clib/exninfo.ml", line 65, characters 2-15
frame @ file "toplevel/vernac.ml", line 117, characters 6-19
frame @ file "toplevel/vernac.ml", line 100, characters 6-89
frame @ file "stm/stm.ml", line 3121, characters 11-50
raise @ file "clib/option.ml", line 50, characters 9-21
```

**Comment by maximedenes** [comment]
Is that after #9530?

**Comment by ejgallego** [comment]
No, it is on master.

**Comment by maximedenes** [comment]
This has been fixed by #9530.
